### ISSUE TITLE
Invite you to try a new series of Lex/Yacc tools

### ISSUE THREAD

**Issue by xp44mm** [fsprojects/FsLexYacc#151]
Lex/Yacc programming enthusiasts, including the maintainers of this repo, I am very pleased to invite you to try a new series of lex/yacc tools. It is a very powerful, easy-to-use next-generation series of Lex/Yacc tools.

The open source see to:  [xp44mm/FSharpCompiler](https://github.com/xp44mm/FSharpCompiler)

The usage of `FSharpCompiler.Lex` and `FSharpCompiler.Analyzing`, see the resolution: [xp44mm/PolynomialExpressions](https://github.com/xp44mm/PolynomialExpressions)

The usage of `FSharpCompiler.Yacc` and `FSharpCompiler.Parsing`, see the resolution: [xp44mm/ArithmeticExpressions](https://github.com/xp44mm/ArithmeticExpressions)

**Comment by sergey-tihon** [comment]
@baronfel

**Comment by xp44mm** [comment]
Why use these packages:

- The Lex/Yacc input file is independent of any programming language, and for each rule, the input file has set a preset semantic action.
- The Lex/Yacc generator converts human-readable input files into a data-driven table that drive the lexical analyzer or the parser.
- The generated lexical analyzer or parser is eventually combined into a pure function with no side effects. This function resolves the input sequence of tokens to become the proper data structure. according to the regular expression in Lex input or according to the free-context grammar in Yacc input.
- You can use a functional approach to repeatedly apply the generated lexical analyzer or parser to process the data stream.

**Comment by xp44mm** [comment]
`FSharpCompiler` is a generic compiler generator written in F# language. 

Its rules do not contain semantic actions, so it is programming language independent. you can use any language you are familiar with to write your compiler.

You can recursively apply regular expressions or context-free grammar to syntax sequences using combination approachs. Using it is as easy as `Regex`.

**Comment by kkm000** [comment]
@xp44mm, and how does it compare to FParsec?

**Comment by xp44mm** [comment]
@kkm000, The FSharpCompiler, It is the compiler of lex/yacc syntax, more like fslexyacc.

**Comment by kkm000** [comment]
@xp44mm, I see. On the surface, yacc looks like yet another yacc (yayacc, given that "yacc" stands for "yet another compiler compiler"). What's the principal difference?

In the lexer, I could not understand what do you mean by "combination approach" (this is why I asked about FParsec, even before looking). It does not look like a library of combinators.

**Comment by xp44mm** [comment]
"combination approach" refers to the combination of lex-generated analyzer or yacc-generated parser. while, the traditional compiler can only be used once in `lex-> yacc->ast`.

**Comment by kkm000** [comment]
@sergey-tihon @baronfel As far as parsing combinators go, I just discovered https://teo-tsirpanis.github.io/Farkle/. I have not played it it yet. Wondering if you know and how good it is. Recently FParsec was my go-to parser (e.g., I used one for the [ABNF form defined by WWWC for SRGS grammars](https://www.w3.org/TR/speech-grammar/)). I like it (or maybe just get used to it), but explicitly accepting whitespace may get annoying. I had to define 3 combinators for whitespace in different contexts (e.g., in a file header linefeed is significant; in others any whitespace including LF is optional and in yet others it's not). A source of confusion and errors, indeed.
